integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/LuceneExtensionIT.java

```diff
@@ -255,10 +261,9 @@ public void query_explain_multipleIndexes_missingIndexName() {
 				() -> query.explain( FIRST_ID )
 		)
 				.isInstanceOf( SearchException.class )
-				.hasMessageContaining( "explain(String id) cannot be used when the query targets multiple indexes" )
-				.hasMessageContaining(
-						"pass one of [" + INDEX_NAME + ", " + OTHER_INDEX_NAME + "]"
-				);
+				.hasMessageContaining( "explain(String id) cannot be used when the query targets multiple mapped types" )
+				.hasMessageContaining( TYPE_NAME )
+				.hasMessageContaining( OTHER_TYPE_NAME );
```

Comment:
I still don't get why you're no longer checking whether the string contains `"pass one of [" + INDEX_NAME + ", " + OTHER_INDEX_NAME + "]"`?

Reply:
Differently from Elasticsearch, the class `LuceneScopeModel` stores `typeNames` and `indexNames` as `HashSet`s and not as `LinkedHashSet`. So I don't think the order with which they appear is predictable. May I have to change their implementations?

Reply:
> May I have to change their implementations?

Yes, definitely. You should change the implementation to use `LinkedHashSet` and behave predictably, not change the tests to relax the constraints :)

Reply:
I will do

Reply:
changed